webbpsf/optics.py

```diff
@@ -2,17 +2,284 @@
+            'A2-2': (1.146185, 0.661750),
+            'A3-3': (1.146185, -0.661750),
+            'A4-4': (0.000000, -1.323500),
+            'A5-5': (-1.146185, -0.661750),
+            'A6-6': (-1.146185, 0.661750),
+            'B1-7': (0.000000, 2.634719),
+            'B2-9': (2.281734, 1.317360),
+            'B3-11': (2.281734, -1.317359),
+            'B4-13': (0.000000, -2.634719),
+            'B5-15': (-2.281734, -1.317360),
+            'B6-17': (-2.281734, 1.317360),
+            'C1-8': (1.142963, 1.979670),
+            'C2-10': (2.285926, 0.000000),
+            'C3-12': (1.142963, -1.979670),
+            'C4-14': (-1.142963, -1.979670),
+            'C5-16': (-2.285926, -0.000000),
+            'C6-18': (-1.142963, 1.979670)}
+
+
+
+    def get_transmission(self,wave):
+
+        segpaths = {}
+        strutpaths=[]
+        for segname, vertices in self.segdata:
+            segpaths[segname] = matplotlib.path.Path(vertices)
+        for strutname, vertices in self.strutdata:
+            strutpaths.append(matplotlib.path.Path(vertices))
+
+        y,x = wave.coordinates()
+        pts = np.asarray([a for a in zip(x.flat, y.flat)])
+        npix = wave.shape[0]
+        out = np.zeros((npix,npix))
+
+        # paint the segments 1 but leave out the SMSS struts
+        for segname, p in segpaths.items():
+            res = p.contains_points(pts)
+            res.shape=(npix,npix)
+            out[res]=1 if not self.label_segments else int(segname.split('-')[1])
+        for p in strutpaths:
+            res = p.contains_points(pts)
+            res.shape=(npix,npix)
+            out[res]=0
+        return out
+
+
+# Note - the following is **NOT USED YET **
+# This will be finished up and used in a subsequent release to 
+# apply the OTE field dependence. For now just the fixed per SI stuff
+# is there.
+class JWST_OTE_Pupil(poppy.FITSOpticalElement):
+    """The complex OTE pupil, including:
+        1) the aperture geometry, based on the cryo ICD detailed coordinates
+        2) high spatial frequency WFE from the as-built mirrors in Rev G optical model
+        3) mid frequencies from Rev W optical budget
+        4) low frequency field-dependent WFE from the Rev G optical model.
+
+        Parameters
+        -----------
+        level : '
+    """
+    def __init__(self, instrument=None, level='requirements', opd_index=0, **kwargs):
+
+        if instrument is not None:
+            self.instrument = instrument
+            self.instr_name = instrument.name
+            self.tel_coords = instrument._tel_coords()
+        else:
+            self.instrument = None
+            self.instr_name = "NIRCam"
+            # figure out default V2V3 coords here
+            self.tel_coords = (0,0) #?
+
+        # determine filename for pupil amplitude array
+        aperture_file = '../jwst_pupil_revW_npix1024.fits.gz'
+        aperture_file =  os.path.abspath(os.path.join(self._datapath,aperture_file))
```

Comment:
This uses an OS-dependent path separator and walks up the tree from the `self._datapath`. I'd change it to:

```python
aperture_file = os.path.abspath(os.path.join(self._WebbPSF_basepath, 'jwst_pupil_revW_npix1024.fits.gz'))
```